### ISSUE TITLE
twisted/python/runtime.py python version string simplification

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4532]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4532|
|Type|enhancement|
|Created|2010-06-30 12:07:57Z|
|Branch|https://github.com/twisted/twisted/tree/simpler-short-version-4532|

From Chris Rebert:

```
> I believe one of the functions in [runtime.py] can be rewritten to be
> significantly simpler/shorter:
> 
> def shortPythonVersion():
>     return "%s.%s.%s" % sys.version_info[:3]
```

Attachments:

* [shortPythonVersion.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c9c/c9c9f45418dfa0f038eb3b30c3e77c0ac136f3e6/20c2cca663d33bc6fdabb959ac1f88bd0f4556e2.patch) (1368 bytes) - added by vperic on 2012-05-26 14:13:46Z - 
* [shortPythonVersion2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c9c/c9c9f45418dfa0f038eb3b30c3e77c0ac136f3e6/203e1549c89332804210320a7a637084ba860771.patch) (1373 bytes) - added by vperic on 2012-05-26 14:49:13Z - Had a typo in the first patch.

<details><summary>Searchable metadata</summary>

```
trac-id__4532 4532
type__enhancement enhancement
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__branches_simpler_short_version_4532 branches/simpler-short-version-4532
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1277899677000000 1277899677000000
changetime__1338110587000000 1338110587000000
version__None None
owner__therve therve
cc__thijs cc__vperic
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

`shortPythonVersion` is also missing unit tests.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_3"></a>|@vperic commented|
|-|-|

Here's a simple patch for this.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

(In [34453]) Branching to 'simpler-short-version-4532'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @therve|
|-|-|

Looks great!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [34457]) Merge simpler-short-version-4532

Author: vperic
Reviewer: therve
Fixes: [#4532](https://github.com/twisted/twisted/issues/4532)

Use sys.version_info to build twisted.python.runtime.shortPythonVersion instead
of sys.hexversion.
